SWIG/options.i

```diff
@@ -806,6 +806,27 @@ class FdBlackScholesVanillaEnginePtr : public boost::shared_ptr<PricingEngine> {
     }
 };
 
+%{
+using QuantLib::FdBatesVanillaEngine;
+typedef boost::shared_ptr<PricingEngine> FdBatesVanillaEnginePtr;
+%}
+
+%rename(FdBatesVanillaEngine) FdBatesVanillaEnginePtr;
+class FdBatesVanillaEnginePtr : public boost::shared_ptr<PricingEngine> {
+  public:
+    %extend {
+        FdBatesVanillaEnginePtr(const BatesProcessPtr& process,
```

Comment:
This can't work.  `FdBatesVanillaEnginePtr` takes a pointer to `BatesModel`, not to `BatesProcess`.